File: PRACTICA8/Ejercicio.py
class A:
    def __init__(self, x, z):
        self.x = x
        self.z = z
    def incrementaxz(self):
        self.x += 1
        self.z += 1
    def incrementaz(self):
        self.z += 1

class B:
    def __init__(self, y, z):
        self.y = y
        self.z = z
    def incrementayz(self):
        self.y += 1
        self.z += 1
    def incrementaz(self):
        self.z += 1

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x, z)
        B.__init__(self, y, z)
    def incrementaxyz(self):
        self.x += 1
        self.y += 1
        self.z += 1
    def mostrar(self):
        print(f"x:{self.x}, y:{self.y}, z:{self.z}")

objeto = D(6, 5, 2)
objeto.incrementaxyz()
objeto.mostrar()
objeto.incrementayz()
objeto.mostrar()

File: PRACTICA8/Ejercicio.java
class A {
    public int x;
    public  int z;

    public A(int x, int z) {
        this.x = x;
        this.z = z;
    }

    public void incrementaxz() {
        x += 1;
        z += 1;
    }

    public void incrementaz() {
        z += 1;
    }
}

class B {
    protected int y;
    protected int z;

    public B(int y, int z) {
        this.y = y;
        this.z = z;
    }

    public void incrementayz() {
        y += 1;
        z += 1;
    }

    public void incrementaz() {
        z += 1;
    }
}

class D extends A {
    private int y;

    public D(int x, int y, int z) {
        super(x, z);
        this.y = y;
    }

    public void incrementaxyz() {
        x += 1;
        y += 1;
        z += 1;
    }

    public void incrementayz() {
        y += 1;
        z += 1;
    }

    public void mostrar() {
        System.out.println("x:" + x + ", y:" + y + ", z:" + z);
    }
}

public class Main {
    public static void main(String[] args) {
        D objeto = new D(6, 5, 2);
        objeto.incrementaxyz();
        objeto.mostrar();
        objeto.incrementayz();
        objeto.mostrar();
    }
}
